docs/how-to/get-package-source.rst

```diff
@@ -1,194 +1,241 @@
-.. _get-package-source:
-
-===========================
-Get the Source of a Package
+Get the source of a package
 ===========================
 
-.. caution::
-
-    Work in progress!
-
-Before you can work on a packge you need to get the source code of a given 
-package. This article presents multiple ways how you can achive this.
+Before you can work on a :term:`source package<Source Package>` you need to get
+the :term:`source code<Source Code>` of that package. This article presents
+four ways to achieve this: :command:`git-ubuntu`, :command:`pull-pkg`, and
+:command:`apt-get source`, and :command:`dget`.
 
-``git-ubuntu``
---------------
+git-ubuntu
+----------
 
 .. note::
 
-   ``git-ubuntu`` is the modern way of working with ubuntu source packages.
+   ``git-ubuntu`` is the modern way of working with :term:`Ubuntu`
+   source packages.
 
 .. warning::
 
-    ``git-ubuntu`` is still in active development. It is possible that you 
-    run into rough edges or unsupported edge cases. You can ask for help
-    in the #ubuntu-devel channel or open a bug report on launchpad.
+    ``git-ubuntu`` is still in active development and these instructions
+    will likely change over time. While ``git-ubuntu`` will become the
+    default packaging method, for now you may encounter rough edges or
+    unsupported edge cases. You can ask for help in the ``#ubuntu-devel``
+    channel or `open a bug report <GitUbuntuBugs_>`_ on :term:`Launchpad`.
+    Bug reports are very welcome!
+
+Install
+~~~~~~~
+
+The following command will install ``git-ubuntu``:
+
+.. code-block:: bash
+
+    sudo snap install --classic --edge git-ubuntu 
 
-Install::
+Basic usage
+~~~~~~~~~~~
 
-    $ sudo snap install --classic --edge git-ubuntu 
+To clone a source package git repository to a directory:
 
-Basic usage::
+.. code-block:: bash
 
-    # Clone a source package git repository to a directory
-    $ git ubuntu clone PACKAGE [DIRECTORY]
+    git ubuntu clone PACKAGE [DIRECTORY]
 
-    # Generate the orig tarballs for a given source package
-    $ git-ubuntu export-orig
 
-Example::
+To generate the orig tarballs for a given source package:
 
-    $ git ubuntu clone hello 
-    $ cd hello
-    $ git ubuntu export-orig
+.. code-block:: bash
+    git-ubuntu export-orig
+
+Example
+~~~~~~~
+
+.. code-block:: bash
+
+    git ubuntu clone hello 
+    cd hello
+    git ubuntu export-orig
 
 You can find further information in these two blog articles (note that they are from 2017):
 
-- `git ubuntu clone <GitUbuntuBlogArticle1_>`_
-- `Git Ubuntu: More on the imported repositories <GitUbuntuBlogArticle2_>`_
+- `git ubuntu clone <https://ubuntu.com/blog/git-ubuntu-clone>`_
+- `Git Ubuntu: More on the imported repositories <https://ubuntu.com/blog/git-ubuntu-more-on-the-imported-repositories>`_
```

Comment:
I'm a bit reluctant to direct people to 6-year-old blog posts which are only going to fall further and further out of date. Do you know of anything more up to date that we can point to (in the absence of docs, which I know aren't ready yet).

I think we should at least move these down the bottom of the page in a "Further reading" section if we really want to include them. For now, I'd also suggest putting the git-ubuntu section at the bottom, otherwise we're essentially prioritising it even though it's not "The Way" yet. Once it becomes The Way, we can always put it back at the top, or give it its own page.

Reply:
> I'm a bit reluctant to direct people to 6-year-old blog posts which are only going to fall further and further out of date. Do you know of anything more up to date that we can point to (in the absence of docs, which I know aren't ready yet).

Unfortunately this is the most recent stuff I could find :/ I totally share your dissatisfaction.

Reply:
I'm not sure really what the solution is here. I guess we can leave these as they are, but I'll make a note for myself to talk to Robie about getting some sort of blog post or more up-to-date article about git-ubuntu out there.